commands/delete-registry.ts

```diff
@@ -0,0 +1,127 @@
+ */
+export async function deleteRegistry(context?: AzureRegistryNode) {
+    if (!context) {
+        deleteRegistryNoContext();
+        return;
+    }
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: true,
+        placeHolder: 'No',
+        value: 'No',
+        prompt: 'Are you sure you want to delete this registry and its associated images? Enter Y or N: '
+    };
+    let answer = await vscode.window.showInputBox(opt);
+    if (answer == 'N' || answer == 'n') { return };
+
+    let azureAccount = context.azureAccount;
+    if (!azureAccount) {
+        return;
+    }
+
+    if (azureAccount.status === 'LoggedOut') {
+        return;
+    }
+
+    const registries = await AzureCredentialsManager.getInstance().getRegistries();
+    let resourceGroup: string;
+    let subscriptionId: string;
+    for (let i = 0; i < registries.length; i++) {
+        if (registries[i].name === context.registry.name) {
+            resourceGroup = registries[i].id.slice(registries[i].id.search('resourceGroups/') + 'resourceGroups/'.length, registries[i].id.search('/providers/'));
+            subscriptionId = context.subscription.id; //registries[i].id.slice('subscriptions/'.length, registries[i].id.search('/resourceGroups/'));
+            break;
+        }
+    }
+    if (!resourceGroup || !subscriptionId) {
+        throw 'Something went wrong, this registry may no longer exist'
+    }
+    const subs = AzureCredentialsManager.getInstance().getFilteredSubscriptionList();
+    const subscription = subs.find(function (sub): boolean {
+        return sub.subscriptionId === subscriptionId;
+    });
+    const client = AzureCredentialsManager.getInstance().getContainerRegistryManagementClient(context.subscription);
+    await client.registries.beginDeleteMethod(resourceGroup, context.registry.name).then(function (response) {
+        console.log("Success!", response);
+    }, function (error) {
+        console.error("Failed!", error);
+    })
+
+}
+
+/**
+ * this is called if the command is called through the input bar as opposed to on an AzureRegistryNode
+ */
+async function deleteRegistryNoContext() {
+
+    let azureAccount = await AzureCredentialsManager.getInstance().getAccount();
+    let registries = await AzureCredentialsManager.getInstance().getRegistries();
+    let reg: string[] = [];
+    for (let i = 0; i < registries.length; i++) {
+        reg.push(registries[i].name);
+    }
+    let desired = await vscode.window.showQuickPick(reg, { 'canPickMany': false, 'placeHolder': 'Choose a Registry to delete' });
+    if (desired === undefined) throw 'user Exit';
+    let registry = registries.find(reg => { return desired === reg.name });
+
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: true,
+        placeHolder: 'No',
+        value: 'No',
+        prompt: 'Are you sure you want to delete this registry and its associated images? Enter Yes or No: '
```

Comment:
Maybe Enter Yes to confirm would be more accurate, as anything else should reject